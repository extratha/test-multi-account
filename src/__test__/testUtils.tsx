import { RenderOptions, RenderResult, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SWRConfig } from 'swr';
import { theme } from '@/config/config-mui/theme';
import { ThemeProvider as MUIThemeProvider } from '@mui/material/styles';
import { NextIntlClientProvider } from 'next-intl';
import commonTh from '../../public/locales/th/common.json';
import commonEn from '../../public/locales/th/common.json';
type RenderWithProvider = (
  elm: React.ReactElement,
  renderOptions?: ProviderOptions,
) => RenderResult;

export type ProviderOptions = {
  formOptions?: Record<string, unknown>;
} & RenderOptions;

const messages = {
  ...commonTh,
  ...commonEn,
};

const renderWithProviders: RenderWithProvider = (
  component,
  renderOptions?,
  locale = 'th',
) => {
  const wrapper = ({ children }: { children: React.ReactNode }) => (
    <NextIntlClientProvider locale={locale} messages={messages}>
      <MUIThemeProvider theme={theme}>
        <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
      </MUIThemeProvider>
    </NextIntlClientProvider>
  );

  return render(component, {
    wrapper,
    ...renderOptions,
  });
};

export { renderWithProviders as render, userEvent };
export * from '@testing-library/react';
