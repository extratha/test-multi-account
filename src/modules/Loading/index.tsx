import { Container, Stack } from "@mui/material";

import Spinner from "@/components/Spinner";

const Loading = () => {
  return (
    <Container maxWidth="sm" sx={{ px: 4, height: "100vh" }}>
      <Stack justifyContent="center" alignItems="center" height="80%">
        <Spinner />
      </Stack>
    </Container>
  );
};

export default Loading;
