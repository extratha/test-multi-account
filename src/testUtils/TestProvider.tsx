import { ThemeProvider } from "@emotion/react";
import { ReactNode } from "react";
import { SWRConfig } from "swr";

import ModalLayer from "@/components/Modal/ModalLayer";
import createMuiTheme from "@/config/muiTheme/muiTheme";

interface TestProviderProps {
  children?: ReactNode;
}

export const TestProvider = ({ children }: TestProviderProps) => {
  return (
    <ThemeProvider theme={createMuiTheme()}>
      <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
      <ModalLayer />
    </ThemeProvider>
  );
};
