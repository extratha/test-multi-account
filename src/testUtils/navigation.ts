import * as NextNavigation from "next/navigation";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
  useParams: jest.fn(),
  usePathname: jest.fn(),
  useSearchParams: jest.fn(),
  useServerInsertedHTML: jest.fn(),
}));

export interface SpyUseSearchParams {
  get: jest.Mock;
}

export interface SpyUseRouter {
  replace: jest.Mock;
  push: jest.Mock;
  back: jest.Mock;
}

export const spyUseParams = (params: any = {}) => {
  jest.spyOn(NextNavigation, "useParams").mockReturnValue(params);
  return NextNavigation.useParams as jest.Mock;
};

export const spyUseRouter = (): SpyUseRouter => {
  const replace = jest.fn();
  const push = jest.fn();
  const back = jest.fn();
  jest.spyOn(NextNavigation, "useRouter").mockReturnValue({ replace, push, back } as any);
  return { replace, push, back };
};

export const spyUsePathname = (pathname: string) => {
  jest.spyOn(NextNavigation, "usePathname").mockReturnValue(pathname);
};

export const spyUseSearchParams = (): SpyUseSearchParams => {
  const get = jest.fn();
  jest.spyOn(NextNavigation, "useSearchParams").mockReturnValue({ get } as any);
  return {
    get,
  };
};

export const spyUseServerInsertedHTML = () => {
  return jest.spyOn(NextNavigation, "useServerInsertedHTML");
};
