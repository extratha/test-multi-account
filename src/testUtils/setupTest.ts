import "@testing-library/jest-dom";
import "jest-canvas-mock";

import "@/__mocks__/Markdown";
import "@/__mocks__/MockFirebase";
import "@/__mocks__/NextImage";
import { LOCALE_LANGUAGE } from "@/config";
import { initI18next } from "@/locales/i18n";
import locale from "@/locales/th.json";

initI18next(LOCALE_LANGUAGE.TH, locale);

jest.mock("@/api/api", () => {
  const actualModule = jest.requireActual("@/api/api");
  return { ...actualModule, __esModule: true };
});

jest.mock("@/api/apiUnauthorize", () => {
  const actualModule = jest.requireActual("@/api/apiUnauthorize");
  return { ...actualModule, __esModule: true };
});

jest.mock("next/config", () => () => ({
  setConfig: jest.fn(),
  publicRuntimeConfig: {
    apiToken:
      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE3MDY2NzUzNDQsImlhdCI6MTcwNjU4ODk0NCwicHJvZmlsZUlkIjoiZDU1YWY3NDUtOTYyZS00MTk2LTkyOTEtZDg2YTliYWNiZWY2In0.sNgwYjIGLYEVAptqILGgRVGXwt9FCPKHRubEZyaodhg",
    mockBaseApiUrl: "https://www.mock-playground-dashboard.co.th",
    baseApiUrl: "https://www.mock-playground-dashboard.co.th",
    globalEndpoint: "playground-dashboard",
    publicApiHost: "http://localhost:3000/",
  },
}));

jest.mock("cookies-next", () => ({
  __esModule: true,
  ...jest.requireActual("cookies-next"),
  getCookie: jest.fn().mockReturnValue("MOCK_ACCESS_TOKEN"),
}));
