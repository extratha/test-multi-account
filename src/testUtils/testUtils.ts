import { act, render as testRender } from "@testing-library/react";
import dayjs from "dayjs";
import flushPromises from "flush-promises";
import { ReactElement } from "react";

import { TestProvider } from "./TestProvider";

export const API = {
  LOGIN: "/auth/login",
  EXAMPLE_AI_INTERPRET_URL: "/lab/examples/labId",
  GET_LAB_INTERPRETS_ID_URL: "/lab/interprets/labId",
  SUBMIT_LAB_INTERPRETS_URL: "/lab/interprets",
  CONSENT: "/consents",
  TERMS_AND_CONDITIONS: "/consents/terms-conditions",
  TERMS_AND_CONDITIONS_LATEST: "/consents/terms-conditions/latest",
  PRIVACY_POLICY: "/consents/privacy-policies",
  PRIVACY_POLICY_LATEST: "/consents/privacy-policies/latest",
  PUBLIC_TERMS_CONDITIONS: "/consents/terms-conditions/public",
  PUBLIC_PRIVACY_POLICIES: "/consents/privacy-policies/public",
};

export const render = (ui: ReactElement) => {
  return testRender(ui, { wrapper: TestProvider });
};

export const flushPromise = async () => {
  await act(async () => {
    await flushPromises();
  });
};

export const advanceTimersByTime = async (time: number) => {
  await act(async () => {
    jest.advanceTimersByTime(time);
  });
};

export const setMockDate = (date: string) => {
  jest.setSystemTime(dayjs(date).valueOf());
};
