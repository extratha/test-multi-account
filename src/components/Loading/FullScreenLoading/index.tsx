"use client";

import { Modal as MuiModal, Stack, styled } from "@mui/material";
import Image from "next/image";
import { usePathname } from "next/navigation";
import { useEffect } from "react";

import { ImageLoadingStack } from "@/assets";
import { usePageLoadingStore } from "@/store";

const Modal = styled(MuiModal)({
  display: "flex",
  justifyContent: "center",
  alignItems: "center",
});

const ImageSection = styled(Stack)({
  outline: "none",
});

const FullScreenLoading = () => {
  const { isPageLoading, setPageLoading } = usePageLoadingStore();
  const pathname = usePathname();

  useEffect(() => {
    setPageLoading(false);
  }, [pathname]);

  return (
    <>
      {isPageLoading && (
        <Modal data-testid="full-screen-loading" open>
          <ImageSection>
            <Image
              data-testid="full-screen-loading-image"
              alt="full-screen-loading-image"
              width={100}
              height={100}
              src={ImageLoadingStack}
            />
          </ImageSection>
        </Modal>
      )}
    </>
  );
};

export default FullScreenLoading;
