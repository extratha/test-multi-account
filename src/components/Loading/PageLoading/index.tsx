'use client'
import { Box, CircularProgress, Stack, Typography } from '@mui/material';
import Backdrop from '@mui/material/Backdrop';
import Image from 'next/image';
import { usePageLoadingStore } from '@/store';
import { ImageLoadingStack } from '@/assets';
import { usePathname } from 'next/navigation';
import { useEffect } from 'react';

const PageLoading = () => {
  const { isPageLoading, setPageLoading } = usePageLoadingStore();
  const pathname = usePathname()
  useEffect(() => {
    setPageLoading(false)
  }, [pathname])
  return (
    <span>
      <Backdrop
        data-testid="backdrop-loading"
        sx={{
          zIndex: (theme) => theme.zIndex.tooltip + 1,
        }}
        open={isPageLoading}
      >
        <Image data-testid="img-loading-stack" alt='' src={ImageLoadingStack} style={{ width: '70px', height: '70px' }}></Image>
      </Backdrop>
    </span>
  );
};

export default PageLoading;
