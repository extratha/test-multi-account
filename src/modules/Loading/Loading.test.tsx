import { render } from "@/testUtils/testUtils";
import Loading from ".";

describe("LoadingModule", () => {
  it("should render correctly", async () => {
    const { asFragment } = render(<Loading />);
    expect(asFragment()).toMatchSnapshot();
  });
});
