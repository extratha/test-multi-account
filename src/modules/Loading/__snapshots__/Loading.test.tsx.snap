// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingModule should render correctly 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-1 {
  0% {
    stroke-dasharray: 1px,200px;
    stroke-dashoffset: 0;
  }

  50% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -15px;
  }

  100% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -125px;
  }
}

.emotion-0 {
  width: 100%;
  margin-left: auto;
  box-sizing: border-box;
  margin-right: auto;
  display: block;
  padding-left: 16px;
  padding-right: 16px;
  padding-left: 32px;
  padding-right: 32px;
  height: 100vh;
}

@media (min-width:600px) {
  .emotion-0 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:600px) {
  .emotion-0 {
    max-width: 600px;
  }
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 80%;
}

.emotion-2 {
  display: inline-block;
  color: #2196F3;
  -webkit-animation: animation-0 1.4s linear infinite;
  animation: animation-0 1.4s linear infinite;
}

.emotion-3 {
  display: block;
}

.emotion-4 {
  stroke: currentColor;
  stroke-dasharray: 80px,200px;
  stroke-dashoffset: 0;
  -webkit-animation: animation-1 1.4s ease-in-out infinite;
  animation: animation-1 1.4s ease-in-out infinite;
}

<div
    class="MuiContainer-root MuiContainer-maxWidthSm emotion-0"
  >
    <div
      class="MuiStack-root emotion-1"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary emotion-2"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg emotion-3"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate emotion-4"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
